### ISSUE TITLE
Bug: Text table chart, User Permission,  internal Hive, Create Dashboard(relationship, join) 

### ISSUE THREAD

**Issue by wm9947** [metatron-app/metatron-discovery#1846]
Metatron version: final
datasource: csv

1. Text table chart: color setting is not properly work
  custom color is not applied

2. Permission
  If userA have watcher permission on Workspace1, I think userA only can watch on Workspace1.
As I guess, userA cannot make any workbook and Workbench.
But userA can move items from userA's workbook to Workspace1.
Is this bug? or function?

3. internal Hive
  When I use snapshot and Staging DB, I think it use internal Hive which reference from Metatron Menual.
But It makes error message "Hive not configured"
Do I have to do something to use snapshot function (such as install Hive myself or else)?
I try to use Metatron with not configured Hive which means not installed any installed Hive at same position, and with Hive at same position. But both condition is not work.

4. Create Dashboard
 When I try to make a Dashboard and select dataset, I can make a connection between dataset. I feels like configuration a foreign key.
  1) After this action, I only can use single table to make a visualization result.
  2) After this, I wish to make a Join, But I saw the error message.
![image](https://user-images.githubusercontent.com/19628618/55924445-ddc94f00-5c44-11e9-88f9-4d53cb26d3a1.png)

**Comment by wm9947** [comment]
Addtional bug(question)

2. 
If userA has a permission on Dataset1 but Workspace1 do not have permission to Dataset1.
I can reasonably understand move my visualization result to Workspace1(like publishing), but currently Workspace1 can see the original table of the Dataset1 by moved workbook without dataset permission.

**Comment by kyungtaak** [comment]
@wm9947 There are a lot of issues. Could you tell us more in detail on our slack channel?
[invitation link](https://join.slack.com/t/metatron-app/shared_invite/enQtNTAwNjAyNzcwMzIyLTY4NGY5MzliZjQyNjdmMjM2ZjM4MzgwMDZiZWViYTMyNjQzZTQ2NmVjNTdhYmZhYzNmMDY1NzdlYTdkYjIxYjk)

**Comment by i1befree** [comment]
@wm9947 I can't reproduce Number 2 because watcher cannot select move button like this :

<img width="1919" alt="nobutton" src="https://user-images.githubusercontent.com/526622/56013210-96b58980-5d2a-11e9-9aea-d6f8a9f4029a.png">

**Comment by i1befree** [comment]
@wm9947 I think custom color setting works well. Of course, we need to look more at the matching of values ​​and colors.
<img width="1550" alt="custom-color-setting" src="https://user-images.githubusercontent.com/526622/56013379-602c3e80-5d2b-11e9-8626-c5358d49d19c.png">

**Comment by i1befree** [comment]
@kyungtaak Would explain staging db concept? @wm9947 As I know, you should configure staging db.

**Comment by i1befree** [comment]
@wm9947 It seems that there is a bug in the join function of the data source. I also tested and found the following bug. @kyungtaak It seems like a bug.

There were problems
Please check the following messages:
QueryTimeExcetpion: [{"error":"Can not deserialize instance of java.lang.String out of START_OBJECT token\n at [Source: HttpInputOverHTTP@7b515579; line: 1, column: 99] (through reference chain: java.util.ArrayList[0])"}]

**Comment by wm9947** [comment]
@i1befree 
For example,
userA has a manager permission on Workspace1.
userA has a watcher permission on Workspace2.

In this time userA cannot move his workbook, which is on the workspace1, to workspace2.


And the table color,
Once I set the color of the data, I cannot change color.
And seems like your result of the table which is colored background is different with mine.
As you can see below image, it changed text data instead of background.
![image](https://user-images.githubusercontent.com/19628618/56013856-c023e480-5d2d-11e9-90ba-ddc0525152be.png)


@kyungtaak 
Unfortunately, I cannot reach slack channel during my working time. I will try on weekend.

Many thanks,

**Comment by i1befree** [comment]
@wm9947 You're right. If two measure values ​​are present, the range is determined by the first measure value and the maximum and minimum values ​​can not be modified. I think this issue is a bug, and we need to discuss this issue.

**Comment by kyungtaak** [comment]
@i1befree ok! :) First from stagedb.. @wm9947 stagedb is a predefined hive database for internal use within metatron.
It is used as a snapshot repository in the data preparation function and provides additional ingestion options not available in the normal DB, such as supporting partition in the hive when ingesting datasources.

So, to use this, you need to configure the application as below.
```
polaris:
  storage:
    stagedb:
      hostname: your_host
      port: 10000
      username: hive
      password: hive
      metastoreUri: thrift://your_host:9083
```

If you have any problems with the configuration for stagedb, please let me know. :)

**Comment by wm9947** [comment]
@kyungtaak 
I add a configuration which you provided. like...

![image](https://user-images.githubusercontent.com/19628618/56412547-8b7cd380-62bf-11e9-813b-7e36b1d1a878.png)

But it is not work.
Is there anything I have to configure more?

**Comment by wm9947** [comment]
@kyungtaak 
I try to make database, which is 'dataprep', in my Hive.
After this action the error message is changed to 'Query failed'
Here is my current conf.
![image](https://user-images.githubusercontent.com/19628618/56485519-4222c300-650f-11e9-8245-a12666ad1860.png)

Many thanks,

**Comment by alchan-lee** [comment]
Hi @wm9947 
For using staging DB, it only needs the previous configuration.
(As @kyungtaak described,
```
  storage:
    stagedb:
      hostname: your_host
      port: 10000
      username: hive
      password: hive
      metastoreUri: thrift://your_host:9083
```
, prefixed dataprep values are things to consider later.)


But it seems to be your settings have not been applied properly.
The first thing that you need to set is activating the template file within the conf directory.
`mv application-config.template.yaml application-config.yaml`

If you are not on the status I mentioned, please feel free to attach your logs detail :)

**Comment by wm9947** [comment]
@alchan-lee 

I set the application-config file as you mentioned.
(change file name, not use prefixed dataprep value)
But still not work.

{"code":"GB0005","message":"unauthorized","details":"Full authentication is required to access this resource"}

This is my error message. I think something in my hive set wrong way.
Do I have to give a admin permission(hive) to use stagedb?
Just I guess, If I want to make(store) a new table in the Hive, I should give a admin permission.

Many thanks,

**Comment by kyungtaak** [comment]
@wm9947 I'm sorry for the late reply. Can I see which screen occurs the error message?

**Comment by wm9947** [comment]
@kyungtaak 

When I want to make a snapshot from the data flow.
and When I want to see the detail of a snapshot data which is saved in local.
(sorry the screenshot is not currently work in my computer)

Additionally, the manual of the geo server installation has been disappeared from git hub main page.
If I want to use the geo server, do I have to configure in the application-config?
(reason why I asked about this, error message has been appear when I do use geoserver configuration at the yaml file.)
I changed the password xxxx to geoserver. is this right? 

Many thanks,

**Comment by kyungtaak** [comment]
> Additionally, the manual of the geo server installation has been disappeared from git hub main page.
> If I want to use the geo server, do I have to configure in the application-config?
> (reason why I asked about this, error message has been appear when I do use geoserver configuration at the yaml file.)
> I changed the password xxxx to geoserver. is this right?

@wm9947 I'm sorry to the late reply
Geoserver is no longer used.  The process that geoserver is doing is handled by discovery and druid.

**Comment by kyungtaak** [comment]
> @wm9947 I think Number 2 is a bug.

fixed this issue. :)
